libs/vertexai/langchain_google_vertexai/_content_catching.py

```diff
@@ -0,0 +1,75 @@
+from datetime import datetime, timedelta
+from typing import List, Optional
+
+from langchain_core.messages import BaseMessage
+from vertexai.preview import caching  # type: ignore
+
+from langchain_google_vertexai.chat_models import (
+    ChatVertexAI,
+    _parse_chat_history_gemini,
+)
+from langchain_google_vertexai.functions_utils import (
+    _format_to_gapic_tool,
+    _format_tool_config,
+    _ToolConfigDict,
+    _ToolsType,
+)
+
+
+def create_context_cache(
```

Comment:
since it's in a private module, afaik, it won't make it to the public python docs:
https://api.python.langchain.com/en/latest/google_vertexai_api_reference.html

should we move it elsewhere?

Reply:
You are right! I can make it public. I didn't want to put it in `ChatModels` because that file is getting a little bit big. What do you think?

Reply:
maybe it's time to introduce public utils :).